src/Debugger.cpp

```diff
@@ -2888,14 +2888,24 @@ void Debugger::setInitialDebuggerState() {
 
 	const QString filename = sessionFilename();
 	if (!filename.isEmpty()) {
-
+        std::shared_ptr<IBreakpoint> bp;
         const ISessionManager &session_manager = edb::v1::session_manager();
 
         if (Result<void, SessionError> session_error = edb::v1::session_manager().loadSession(filename)) {
+            QVariantList breakpoints_data = session_manager.breakpoints();
             QVariantList comments_data = session_manager.comments();
             QVariantList labels_data = session_manager.labels();
             ui.cpuView->restoreComments(comments_data);
             edb::v1::symbol_manager().restoreLabels(labels_data);
+            for (auto it = breakpoints_data.begin(); it != breakpoints_data.end(); ++it) {
+                IBreakpoint::TypeId type;
+                QVariantMap bp_data = it->toMap();
+
+                type = static_cast<IBreakpoint::TypeId>(bp_data["type"].toInt());
+                bp = edb::v1::create_breakpoint(edb::v1::string_to_address(bp_data["address"].toString()).value());
+                bp_data["enabled"].toBool() ? bp->enable() : bp->disable();
+                //bp->setType(type);
```

Comment:
Is there any problem with this call?

Reply:
When trying this patch, the call to the setType method completely hangs EDB for an unknown reason. Il will look forward into this issue and bring a solution.

Reply:
Fixed in the last force-pushed update.
Note : calling SetType with a IBreakpoint::TypeId causes an infinite loop.